### PR TITLE
calibrated chassis

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

**Review by aaa-png — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 1]
```diff
@@ -21,13 +22,13 @@ public class MK4iSwerveConstants {
 	
 	public static final double ANGULAR_GEAR_RATIO = (150.0 / 7);
 	
-	public static final double LINEAR_GEAR_RATIO = 8.14;
+	public static final double LINEAR_GEAR_RATIO = 6.12;
 
 	public static final double ks = 0.32;
 
-	public static final double kv = 2.5289;
+	public static final double kv = 0;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:29]: What

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 2]
```diff
@@ -21,13 +22,13 @@ public class MK4iSwerveConstants {
...
 
-	public static final double ka = 0.40665;
+	public static final double ka = 0;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:31]: What

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 3]
```diff
@@ -47,24 +48,22 @@ public class MK4iSwerveConstants {
 		PID_CONFIGS.kS = ks;
 		PID_CONFIGS.kA = ka;
 		PID_CONFIGS.kV = kv;
-		PID_CONFIGS.kP = 40;
+		PID_CONFIGS.kP = 20;
 		PID_CONFIGS.kI = 0;
-		PID_CONFIGS.kD = 0;
+		PID_CONFIGS.kD = 1;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:53]: Kd of one is pretty weird

File: src/main/java/edu/greenblitz/robotName/utils/hid/SmartJoystick.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
  */
 public class SmartJoystick {
 	
-	private static final double DEADZONE = 0.05;
+	private static final double DEADZONE = 0.07;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/utils/hid/SmartJoystick.java:17]: Oh no…

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java (hunk 1 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:29]: we are doing position control on the angular swerve motor so we dont need kv

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java (hunk 2 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:31]: we are doing position control on the angular swerve motor so we dont need ka

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java (hunk 3 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:53]: and? this is what we calibrated

File: src/main/java/edu/greenblitz/robotName/utils/hid/SmartJoystick.java (hunk 4 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/utils/hid/SmartJoystick.java:17]: the deadzone was so small it was causing issues so its now bigger

Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/utils/hid/SmartJoystick.java:17]: That means our controllers are getting worse

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java (hunk 3 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:53]: It’s just weird that you had to use kd and not max acceleration

**Review by yonnbro — APPROVED** [review]

**Comment by yonnbro** [comment]
Dis you test it on 1 meter? And 1 meter front and 1 meter right or something

**Comment by YoniKiriaty** [comment]
> Did you test it on 1 meter? And 1 meter front and 1 meter right or something

we did some driving with it and it seemed good

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerve.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+import static edu.wpi.first.units.Units.Volts;
+
+public class SysIdFalconSwerve extends GBSubsystem {
+
+    private static SysIdFalconSwerve instance;
+
+    private TalonFX motorLB;
+    private TalonFX motorRB;
+    private TalonFX motorLF;
+    private TalonFX motorRF;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerve.java:24]: add spaces + the convention is b/f and then l/r

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+public class SysIdFalconSwerveConstants {
+
+    protected static final double SIGNAL_SPEED = 250;
+
+    protected static final double DYNAMIC_VOLTAGE = 0.5;
+
+    protected static final int MOTOR_LB = 43;
+    protected static final int MOTOR_LF = 4;
+    protected static final int MOTOR_RF = 2;
+    protected static final int MOTOR_RB = 6;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java:16]: sapces + b/f before l/r

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    
+    public static final CurrentLimitsConfigs CURRENT_LIMITS_CONFIGS = new CurrentLimitsConfigs();
+    static {
+        CURRENT_LIMITS_CONFIGS.StatorCurrentLimitEnable = true;
+        CURRENT_LIMITS_CONFIGS.StatorCurrentLimit = 40;
+    }
+    
+    public static final TalonFXConfiguration a = new TalonFXConfiguration();
+    static{
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java:37]:
```suggestion
    static {
```
and space

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+    public static final HardwareLimitSwitchConfigs HARDWARE_LIMIT_SWITCH_CONFIGS = new HardwareLimitSwitchConfigs();
+
+    public static final FeedbackConfigs FEEDBACK_CONFIGS = new FeedbackConfigs();
+
+    static {
+        FEEDBACK_CONFIGS.SensorToMechanismRatio = MK4iSwerveConstants.LINEAR_GEAR_RATIO;
+    }
+    
+    public static final CurrentLimitsConfigs CURRENT_LIMITS_CONFIGS = new CurrentLimitsConfigs();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconSwerveConstants.java:30]: sapce

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**